coordinator/src/services/communication/tracker/translation.ts

```diff
@@ -0,0 +1,234 @@
+    case "With Dispenser - Active":
+      return {status: fhir.TaskStatus.IN_PROGRESS, businessStatus: "0003"}
+    case "Expired":
+      return {status: fhir.TaskStatus.FAILED, businessStatus: "0004"}
+    case "Cancelled":
+      return {status: fhir.TaskStatus.CANCELLED, businessStatus: "0005"}
+    case "Dispensed":
+      return {status: fhir.TaskStatus.COMPLETED, businessStatus: "0006"}
+    case "Not Dispensed":
+      return {status: fhir.TaskStatus.COMPLETED, businessStatus: "0007"}
+    default:
+      throw new Error("Unexpected Status Code from Spine")
+  }
+}
+
+function createCourseOfTherapyTypeExtension(treatmentType: string): fhir.ExtensionExtension<fhir.CodingExtension> {
+  let code, display
+  switch (treatmentType) {
+    case "Acute Prescription":
+      code = fhir.CourseOfTherapyTypeCode.ACUTE
+      display = "Short course (acute) therapy"
+      break
+    case "Repeat Prescribing":
+      code = fhir.CourseOfTherapyTypeCode.CONTINUOUS
+      display = "Continuous long term therapy"
+      break
+    case "Repeat Dispensing":
+      code = fhir.CourseOfTherapyTypeCode.CONTINUOUS_REPEAT_DISPENSING
+      display = "Continuous long term (repeat dispensing)"
+      break
+  }
+  return {
+    url: "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-Prescription",
+    extension:  [{
+      url: "courseOfTherapyType",
+      valueCoding: {
+        system: "http://terminology.hl7.org/CodeSystem/medicationrequest-course-of-therapy",
+        code: code,
+        display: display
+      }
+    }]
+  }
+}
+
+function convertToFhirDate(dateString: string) {
+  return moment.utc(dateString, HL7_V3_DATE_TIME_FORMAT).format(ISO_DATE_FORMAT)
+}
+
+function createRepeatInfoExtension(currentIssue: string, totalAuthorised: string) {
+  return {
+    url: "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation",
+    extension: [
+      {
+        url: "numberOfRepeatsAllowed",
+        valueUnsignedInt: totalAuthorised
+      },
+      {
+        url: "numberOfRepeatsIssued",
+        valueUnsignedInt: currentIssue
+      }
+    ]
+  }
+}
+
+function convertLineItemToInput(lineItemId: string, prescription: DetailPrescription) {
+  const lineItem = prescription.lineItems[lineItemId]
+  const taskInput: fhir.TaskInput = {
+    type: fhir.createCodeableConcept("http://snomed.info/sct", "16076005", "Prescription"),
+    valueReference: fhir.createIdentifierReference(
+      fhir.createIdentifier("https://fhir.nhs.uk/Id/prescription-order-item-number", lineItemId.toLowerCase())
+    )
+  }
+
+  const dispensingInformationExtension = []
+  if (prescription.prescriptionDispensedDate && prescription.prescriptionDispensedDate !== "False") {
+    dispensingInformationExtension.push({
+      url: "dateLastDispensed",
+      valueDate: convertToFhirDate(prescription.prescriptionDispensedDate)
+    })
+  }
+
+  if (lineItem.itemStatus) {
+    const {businessStatus} = getStatusCodesFromDisplay(lineItem.itemStatus)
```

Comment:
I think line items have a different set of codes to prescriptions. Might need a separate method for this

Reply:
That will be the reason for the different "To be dispensed" descriptions as well